tests/core/pyspec/eth2spec/test/merge/genesis/test_initialization.py

```diff
@@ -120,3 +120,55 @@ def test_initialize_post_transition(spec):
     assert spec.is_merge_complete(state)
 
     yield 'state', state
+
+
+@with_merge_and_later
+@spec_test
+@single_phase
+@with_presets([MINIMAL], reason="too slow")
+def test_genesis_block_execution_payload(spec):
+    """
+    Test that the merge genesis block includes the execution payload that was
+    used to make the genesis state.
+    """
+    # Create a genesis execution payload header with no transactions
+    genesis_execution_header = get_sample_genesis_execution_payload_header(spec)
+    genesis_execution_header.transactions_root = \
+        spec.List[spec.Transaction, spec.MAX_TRANSACTIONS_PER_PAYLOAD]().hash_tree_root()
+
+    # Initialize data required to make state/block
+    eth1_block_hash = b'\x12' * 32
+    eth1_timestamp = spec.config.MIN_GENESIS_TIME
+    deposit_count = spec.config.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT
+    deposits, deposit_root, _ = prepare_full_genesis_deposits(
+        spec,
+        spec.MAX_EFFECTIVE_BALANCE,
+        deposit_count,
+        signed=True,
+    )
+
+    # Get genesis state
+    genesis_state = spec.initialize_beacon_state_from_eth1(
+        eth1_block_hash,
+        eth1_timestamp,
+        deposits,
+        execution_payload_header=genesis_execution_header
+    )
+
+    # Get genesis block
+    genesis_block = spec.get_genesis_block_from_genesis_state(genesis_state)
+
+    # Test that genesis block and genesis state refer to the same execution payload
+    assert genesis_block.body.execution_payload.hash_tree_root() == \
+        genesis_state.latest_execution_payload_header.hash_tree_root()
+
+    # Check that the payload we generated is the one that is embedded in the state
+    assert genesis_execution_header.hash_tree_root() == \
+        genesis_state.latest_execution_payload_header.hash_tree_root()
+
+    # Check that the genesis block points to the genesis state
+    assert genesis_block.state_root == genesis_state.hash_tree_root()
+
+    # Do a static assert of the genesis block given the parameters above
+    assert "bde449ed275fced558631288addaac3f1dcd6eeb00ddb9fb0c293ea7dcd3b3eb" == \
```

Comment:
We generally always avoid `\` to signify a new line for same logical statement across the repo. Suggest  something like

```python
    expected_genesis_block_root = "bde449ed275fced558631288addaac3f1dcd6eeb00ddb9fb0c293ea7dcd3b3eb"
    assert genesis_block.hash_tree_root().hex() == expected_genesis_block_root

Reply:
I'm also not sure we should hard-code this in here. All this does is ensure that the inputs and `beacon_block` structure don't change since when this test was written. This will almost certainly prevent this test from being useful for subsequent forks because `BeaconBlock` is likely be extended with new fields.

Seems brittle and doesn't buy us much. The relationships between the various components as defined in the above asserts are sufficient, imo

Reply:
we could scope this check to each fork -- the idea is that once we publish the spec that yields this root hash, it *should not* change, and this `assert` is an additional check that nothing does